pkg/controller/daemon/daemoncontroller.go

```diff
@@ -446,7 +448,7 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	}
 	var nodesNeedingDaemonPods, podsToDelete []string
 	for _, node := range nodeList.Items {
-		shouldRun := dsc.nodeShouldRunDaemonPod(&node, ds)
+		shouldRun := dsc.nodeShouldRunDaemonPod(&node, ds, nodeToDaemonPods)
```

Comment:
`nodeToDaemonPods ` is not enough; it only includes the Pods matching ds's selector (or ownReference). If we want to enable PodAffinity/Anti-Affinity, we need to include all Pods on the node.